### PR TITLE
[BROWSEUI] CBandSiteMenu: Show Toolbar error on failure

### PR DESCRIPTION
**Pull request by binarymaster**
## Purpose

Show proper toolbar creation error message, like Windows does.

- :white_check_mark:  Multiple message boxes can be created (same for Windows)
- :white_check_mark:  Only for built-in toolbars (desktop / quick launch, same as Windows)
- :white_check_mark:  Allow the message strings to be translated

## Proposed changes
- Add function to display toolbar creation error message

## TODO (for me)
- Add Russian translation
- Add Polish translation

### CODE REVIEW

**Comment by binarymaster** [comment]
Demonstration:

virtualbox_winxp_07_11_2017_13_22_55
virtualbox_reactos_07_11_2017_13_28_10

Cc @yagoulas

File: dll/win32/browseui/shellbars/CBandSiteMenu.cpp [hunk 1]
```diff
@@ -386,6 +386,21 @@ HRESULT STDMETHODCALLTYPE CBandSiteMenu::QueryContextMenu(
     return MAKE_HRESULT(SEVERITY_SUCCESS, 0, USHORT(idMax - idCmdFirst +1)); 
 }
 
+HRESULT CBandSiteMenu::_ShowToolbarError(HRESULT hRet)
+{
+    WCHAR szText[260];
+    WCHAR szTitle[256];
+
+    if (!LoadStringW(GetModuleHandleW(L"browseui.dll"), IDS_TOOLBAR_ERR_TEXT, szText, _countof(szText)))
+        wcscpy(szText, L"Cannot create toolbar.");
```
Comment (ThFabba) [inline dll/win32/browseui/shellbars/CBandSiteMenu.cpp]: Please don't use this function. You want String(Cb/Cch)Copy(Ex)W or wcscpy_s.

File: dll/win32/browseui/shellbars/CBandSiteMenu.cpp [hunk 2]
```diff
@@ -386,6 +386,21 @@ HRESULT STDMETHODCALLTYPE CBandSiteMenu::QueryContextMenu(
...
+
+    if (!LoadStringW(GetModuleHandleW(L"browseui.dll"), IDS_TOOLBAR_ERR_TITLE, szTitle, _countof(szTitle)))
+        wcscpy(szTitle, L"Toolbar");
+
+    MessageBoxW(NULL, szText, szTitle, MB_ICONSTOP | MB_SETFOREGROUND);
```
Comment (ThFabba) [inline dll/win32/browseui/shellbars/CBandSiteMenu.cpp]: Explicitly adding MB_OK seems like a good idea.

File: dll/win32/browseui/shellbars/CBandSiteMenu.cpp (hunk 1 above)
Comment (binarymaster) [reply dll/win32/browseui/shellbars/CBandSiteMenu.cpp]: Done, thanks!

**Pull request merged**